ykllvmwrap/src/memman.cc

```diff
@@ -5,35 +5,47 @@
 
 using namespace llvm;
 
-MemMan::MemMan() {}
-MemMan::~MemMan() {}
-
-uint8_t *alloc_mem(uintptr_t Size, unsigned Alignment,
+uint8_t *allocateSection(uintptr_t Size, unsigned Alignment,
                    std::vector<AllocMem> *Vec) {
   uintptr_t RequiredSize = Alignment * ((Size + Alignment - 1) / Alignment + 1);
-  auto Ptr = (unsigned char *)mmap(0, RequiredSize, PROT_WRITE,
-                                   MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+  auto Ptr = (uint8_t *) mmap(0, RequiredSize,
+                PROT_READ | PROT_WRITE,
```

Comment:
We added the READ permission here. Why's that again?

Reply:
I think on some systems WRITE doesn't imply READ, but I imagine stackmaps won't work on those systems anyway, so it doesn't really matter too much. Just though it's more accurate this way.

Reply:
FWIW, it's fairly normal to have pages RW, so I'm happy with the change anyway.

Reply:
> I think on some systems WRITE doesn't imply READ

That's correct: even more, some systems (e.g. OpenBSD) cause a segfault if a page is `WRITE` but not `READ`.